test/python/quantum_info/operators/symplectic/test_pauli_list.py

```diff
@@ -0,0 +1,1470 @@
+            pauli = PauliList(["XI", "IZ", "YY"])
+            array = np.array([[False, False], [True, False], [True, True]], dtype=bool)
+            self.assertTrue(np.all(pauli.z == array))
+
+        with self.subTest(msg="set Z"):
+            pauli = PauliList(["XI", "IZ"])
+            val = np.array([[False, False], [True, True]], dtype=bool)
+            pauli.z = val
+            self.assertEqual(pauli, PauliList(["XI", "ZZ"]))
+
+        with self.subTest(msg="set Z raises"):
+            with self.assertRaises(Exception):
+                pauli = PauliList(["XI", "IZ"])
+                val = np.array([[False, False, False], [True, True, True]], dtype=bool)
+                pauli.z = val
+
+    def test_phase_property(self):
+        """Test phase property"""
+        with self.subTest(msg="phase"):
+            pauli = PauliList(["XI", "IZ", "YY", "YI"])
+            array = np.array([0, 0, 0, 0], dtype=int)
+            np.testing.assert_equal(pauli.phase, array)
+
+        with self.subTest(msg="set phase"):
+            pauli = PauliList(["XI", "IZ"])
+            val = np.array([2, 3], dtype=int)
+            pauli.phase = val
+            self.assertEqual(pauli, PauliList(["-XI", "iIZ"]))
+
+        with self.subTest(msg="set Z raises"):
+            with self.assertRaises(Exception):
+                pauli = PauliList(["XI", "IZ"])
+                val = np.array([1, 2, 3], dtype=int)
+                pauli.phase = val
+
+    def test_shape_property(self):
+        """Test shape property"""
+        shape = (3, 4)
+        pauli = PauliList.from_symplectic(np.zeros(shape), np.zeros(shape))
+        self.assertEqual(pauli.shape, shape)
+
+    @combine(j=range(1, 10))
+    def test_size_property(self, j):
+        """Test size property"""
+        shape = (j, 4)
+        pauli = PauliList.from_symplectic(np.zeros(shape), np.zeros(shape))
+        self.assertEqual(len(pauli), j)
+
+    @combine(j=range(1, 10))
+    def test_n_qubit_property(self, j):
+        """Test n_qubit property"""
+        shape = (5, j)
+        pauli = PauliList.from_symplectic(np.zeros(shape), np.zeros(shape))
+        self.assertEqual(pauli.num_qubits, j)
+
+    def test_eq(self):
+        """Test __eq__ method."""
+        pauli1 = PauliList(["II", "XI"])
+        pauli2 = PauliList(["XI", "II"])
+        self.assertEqual(pauli1, pauli1)
+        self.assertNotEqual(pauli1, pauli2)
+
+    def test_len_methods(self):
+        """Test __len__ method."""
+        for j in range(1, 10):
+            labels = j * ["XX"]
+            pauli = PauliList(labels)
+            self.assertEqual(len(pauli), j)
+
+    def test_add_methods(self):
+        """Test __add__ method."""
+        labels1 = ["XXI", "IXX"]
+        labels2 = ["XXI", "ZZI", "ZYZ"]
+        pauli1 = PauliList(labels1)
+        pauli2 = PauliList(labels2)
+        target = PauliList(labels1 + labels2)
+        self.assertEqual(target, pauli1 + pauli2)
+
+    def test_add_qargs(self):
```

Comment:
can test also pseudo-random pauli's

Reply:
Thanks. Why do we need the test for the random Paulis here?